lib/roo/base.rb

```diff
@@ -32,6 +32,14 @@ def initialize(filename, options = {}, _file_warning = :error, _tmpdir = nil)
     @last_column = {}
 
     @header_line = 1
+  rescue => e
```

Comment:
No, it should be `rescue`.

You don't want the connection closed immediately after `initialize` returns.  You only want `initialize` to close the connection if it **doesn't** return.